File: src/main/java/qmech/lib/util/FuelHandler.java
package qmech.lib.util;

import cpw.mods.fml.common.IFuelHandler;
import cpw.mods.fml.common.registry.GameRegistry;
import net.minecraft.item.ItemStack;

import java.util.HashMap;
import java.util.Map;

public class FuelHandler implements IFuelHandler {

    private final Map<ItemStack, Integer> fuels = new HashMap<ItemStack, Integer>();

    public FuelHandler() {
        GameRegistry.registerFuelHandler(this);
    }

    public void addFuel(ItemStack fuel, int time) {
        this.fuels.put(fuel, time);
    }

    @Override
    public int getBurnTime(ItemStack fuelStack) {
        Integer time = this.fuels.get(fuelStack);
        return (time != null) ? time : 0;
    }

}

File: src/main/scala/geek/lawsof/physics/lib/util/handler/FuelHandler.scala
package geek.lawsof.physics.lib.util.handler

import cpw.mods.fml.common.IFuelHandler
import cpw.mods.fml.common.registry.GameRegistry
import net.minecraft.item.ItemStack

/**
 * Created by anshuman on 28-05-2014.
 */
object FuelHandler extends IFuelHandler {
  GameRegistry.registerFuelHandler(this)

  var fuels: Map[ItemStack, Int] = Map.empty

  def add(stack: ItemStack, items: Int) = fuels += (stack -> items * 200)

  def += (fuel: (ItemStack, Int)) = add(fuel._1, fuel._2)

  override def getBurnTime(fuel: ItemStack): Int = fuels(fuel)
}
